pkg/emulator/libretro/image/scale.go

```diff
@@ -11,70 +13,122 @@ const (
 	ScaleOld = 0
 	// nearest neighbour interpolation
 	ScaleNearestNeighbour = 1
+	// bilinear interpolation
+	ScaleBilinear = 2
 )
 
-func Resize(scaleType int, fn Format, w int, h int, packedW int, vw int, vh int, bpp int, data []byte, image *image.RGBA) {
+func Resize(scaleType int, fn Format, w int, h int, packedW int, vw int, vh int, bpp int, data []byte, out *image.RGBA) {
+
+	// !to implement own image interfaces img.Pix = bytes[]
+	src := image.NewRGBA(image.Rect(0, 0, w, h))
+	toRgba(fn, w, h, packedW, bpp, data, src)
+
 	// !to do set it once instead switching on each iteration
 	// !to do skip resize if w=vw h=vh
 	switch scaleType {
 	case ScaleSkip:
-		skip(fn, w, h, packedW, vw, vh, bpp, data, image)
+		skip(fn, w, h, packedW, vw, vh, bpp, data, src, out)
 	case ScaleNearestNeighbour:
-		nearest(fn, w, h, packedW, vw, vh, bpp, data, image)
+		draw.NearestNeighbor.Scale(out, out.Bounds(), src, src.Bounds(), draw.Src, nil)
```

Comment:
OMG golang has built-in support @_@. Thanks for finding that out. Sorry if you took time to re-implement it :-s